WeakAuras/Prototypes.lua

```diff
@@ -142,6 +142,45 @@ WeakAuras.function_strings = {
   ]]
 };
 
+local hsvFrame = CreateFrame("Colorselect")
+
+-- HSV transition, for a much prettier color transition in many cases
+-- see http://www.wowinterface.com/forums/showthread.php?t=48236
+function WeakAuras.GetHSVTransition(perc, r1, g1, b1, r2, g2, b2)
+  --get hsv color for colorA
+  hsvFrame:SetColorRGB(r1,g1,b1)
+  local h1, s1, v1 = hsvFrame:GetColorHSV() --radius, saturation, luminance
+  --get hsv color for colorB
+  hsvFrame:SetColorRGB(r2,g2,b2)
+  local h2, s2, v2 = hsvFrame:GetColorHSV() --radius, saturation, luminance
+  --calculate new HSV values based on HSV values from colorA and colorB and multiply them by the perc
+  local h3 = floor(h1-(h1-h2)*perc)
+  --check if the angle between the two H values is > 180
+  if abs(h1-h2) > 180 then
```

Comment:
Can you explain what is going on here?

Reply:
This is intended to pick the shortest arc of the hue circle, since there's always two options for how to transition from one color to another. If you're going from red to green, for example, you can either pass through yellow or blue.